### DISCUSSION TITLE
是否有意愿将界面的功能分离出来，或则能够兼容其他的界面库例如QT qml

### DISCUSSION THREAD

**Discussion by xiaoyu21git** [godotengine/godot-proposals#8328] [GUI]
### Describe the project you are working on

现在的UI方式其实很难用，因为我是一个QT的忠实用户，我突发奇想如果godot 能想qml一样写UI的代码那该多好？或者将qml的界面介入到godot

### Describe the problem or limitation you are having in your project

UI系统

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

UI布局系统

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

qml 代码

### If this enhancement will not be used often, can it be worked around with a few lines of script?

js脚本

### Is there a reason why this should be core and not an add-on in the asset library?

做成附加组件也可以

**Comment by Zireael07** [comment]
Please use English (e.g. via Google Translate) here.

**Comment by Calinou** [comment]
Note that we don't have plans to integrate Qt in Godot, as Qt is a massive dependency in itself (it's larger than Godot).

**Reply by xiaoyu21git** [reply]
Can you write the UI for godot like you did for QML. Can you consider it? I think the current UI system relies more on controls and its scalability is too weak

**Reply by bruvzg** [reply]
> Qt is a massive dependency in itself 

Also, the open-source version of Qt is GPL/LGPL licensed and therefore unusable on iOS / Android.

**Reply by xiaoyu21git** [reply]
I think we can change the UI system to a form similar to QML, which can greatly improve scalability. If so, perhaps we can make a more cool game UI

**Reply by theraot** [reply]
@xiaoyu21git I can imagine implementing something similar to QML on top of Controls (which would be a hard sell to be a built-in feature, but still possible). However, I do not see how it would be more scalable. Can you elaborate on what QML allows you to do that you find difficult - or impossible - with Controls?

**Comment by mrbbbaixue** [comment]
请使用英语进行交流：）
再者，实际上将Qt和Godot集成几乎不可能。Qt公司可能并不允许界面库随着Godot导出。
